boxoffice/models/discount_policy.py

```diff
@@ -53,6 +58,33 @@ def is_automatic(self):
     def is_coupon(self):
         return self.discount_type == DISCOUNT_TYPE.COUPON
 
+    def gen_signed_code(self):
+        """Generates a signed code in the format discount_code_base.randint.signature"""
+        signer = Signer(self.secret)
```

Comment:
If there's a startup cost to this call, move this to a `@cached_property` method.